Remotion/Core/Core/Collections/DictionaryExtensions.cs

```diff
@@ -27,33 +28,44 @@ namespace Remotion.Collections
   /// </summary>
   public static class DictionaryExtensions
   {
+    [return: MaybeNull]
     public static TValue GetValueOrDefault<TKey, TValue> (this IDictionary<TKey, TValue> dictionary, TKey key)
+        where TKey : notnull

```

Comment:
Not sure about this. The `notnull` contraint on `TKey` was dropped again in .NET Core.

Reply:
The MSDN and sane reasoning are clear: key = null produces an argument null exception. Can you dig up the blamed line of where this was reverted and the associated ticket. we should follow up with Microsoft on this. Until then, notnull it is.

Reply:
okay, there's differening informaiton on the msdn regarding this, even on the same page (Exceptions vs Remarks). So, yes, we stick with the key being allowed to be null when dealing with the idictionary interface.